lib/streamlit/runtime/runtime_util.py

```diff
@@ -62,6 +75,45 @@ def is_cacheable_msg(msg: ForwardMsg) -> bool:
     return msg.ByteSize() >= int(config.get_option("global.minCachedMessageSize"))
 
 
+@overload
+def duration_to_seconds(
+    ttl: float | timedelta | str | None, *, coerce_none_to_inf: Literal[False]
+) -> float | None:
+    ...
+
+
+@overload
+def duration_to_seconds(ttl: float | timedelta | str | None) -> float:
+    ...
+
+
+def duration_to_seconds(
+    ttl: float | timedelta | str | None, *, coerce_none_to_inf: bool = True
+) -> float | None:
+    """
+    Convert a ttl value to a float representing "number of seconds".
```

Comment:
nit: should we rename `ttl` here to `duration` to better match the new function name?

```suggestion
    duration: float | timedelta | str | None, *, coerce_none_to_inf: bool = True
) -> float | None:
    """
    Convert a duration value to a float representing "number of seconds".
```